dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -1277,6 +1274,259 @@ public void flushVanityURLCacheRight() throws ServletException, IOException {
         checkFilter(host_2, vanityUrlUpdated, -1);
     }
 
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Create a ContentType with publish date
+     * - Create two Language
+     * - Create a {@link Contentlet} with the first language and publish date for tomorrow
+     * - Create a {@link Contentlet} with the second language and publish date for after tomorrow
+     * Should: Each {@link Contentlet} be saved and index with the right publish date
+     *
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void differentVersionWithDifferentPublishDate() throws DotDataException, DotSecurityException {
+
+        final Language language1 = new LanguageDataGen().nextPersisted();
+        final Language language2 = new LanguageDataGen().nextPersisted();
+
+
+        final Field publishField = new FieldDataGen().defaultValue(null)
+                .type(DateTimeField.class).next();
+
+        final ContentType contentType = new ContentTypeDataGen()
+                .field(publishField)
+                .publishDateFieldVarName(publishField.variable())
+                .nextPersisted();
+
+        Contentlet contentlet1 = null;
+        Contentlet contentlet2 = null;
+
+        try {
+
+            final Calendar calendar = Calendar.getInstance();
+            calendar.add(Calendar.DATE, 1);
+            final Date tomorrow = calendar.getTime();
+
+            calendar.add(Calendar.DATE, 1);
+            final Date afterTomorrow = calendar.getTime();
+
+            contentlet1 = new ContentletDataGen(contentType)
+                    .setProperty(publishField.variable(), tomorrow)
+                    .languageId(language1.getId())
+                    .nextPersisted();
+
+            contentlet2 = ContentletDataGen.checkout(contentlet1);
+            contentlet2.setLanguageId(language2.getId());
+            contentlet2.setProperty(publishField.variable(), afterTomorrow);
+            ContentletDataGen.checkin(contentlet2);
+
+            checkFromElasticSearch(publishField, tomorrow, afterTomorrow, contentlet1);
+
+            checkFromDataBase(publishField, tomorrow, afterTomorrow, contentlet1);
```

Comment:
why contentlet2 result is never checked out?

Reply:
really the method is checking both

https://github.com/dotCMS/core/blob/9f0e7869e66bb92bf31604b93a882ffc887acd55/dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java#L1412

Reply:
I also found it confusing @nollymar . I think the `checkFromDataBase` method needs java doc or maybe done it in a different way. Basically the method implementation finds the two versions given only contentl1, hence not needing to pass contentlet2 in the params.